src/configFile.ts

```diff
@@ -0,0 +1,73 @@
+import { parse } from "toml";
+import { readFile } from "fs";
+import findUp from "find-up";
+interface foremanConfig {
+  tools: {
+    [tool_name: string]: foremanTool
+  };
+}
+
+interface foremanTool {
+  source?: string;
+  github?: string;
+  gitlab?: string;
+  version: string;
+}
```

Comment:
I was going to ask if we could anticipate the artifactory stuff here, but I think we're less concerned with that since there wouldn't really be fully public sources to pull from.